frontend/src/core/comments/components/AddComment.js

```diff
@@ -25,69 +34,87 @@ export default function AddComments(props: Props) {
-        }
-        else {
-            commentInput.current.rows = maxRows;
-        }
+    function focusEditor() {
+        editor.current.focus();
     }
 
-    const handleOnKeyDown = (event: SyntheticKeyboardEvent<>) => {
-        if (event.keyCode === 13 && event.shiftKey === false) {
-            submitComment(event);
+    function keyBindingFn(e: SyntheticKeyboardEvent<>) {
+        if (e.keyCode === 13 && e.shiftKey === false) {
+          return 'submitOnEnter' 
         }
+        
+        return getDefaultKeyBinding(e)
     }
 
-    const submitComment = (event: SyntheticEvent<>) => {
-        event.preventDefault();
-        const comment = commentInput.current.value;
+    React.useEffect(() => {
+        focusEditor()
+    }, []);
 
-        if (!comment) {
-            return null;
+    const handleKeyCommand = (command: string) => {
+        if (command === 'submitOnEnter') {
+            submitComment();
+            return 'handled';
         }
+        return 'not-handled';
+    }
+
+    const clearEditorContent = () => {
+        let contentState = editorState.getCurrentContent();
+        const firstBlock = contentState.getFirstBlock();
+        const lastBlock = contentState.getLastBlock();
+        const allSelected = new SelectionState({
+            anchorKey: firstBlock.getKey(),
+            anchorOffset: 0,
+            focusKey: lastBlock.getKey(),
+            focusOffset: lastBlock.getLength(),
+            hasFocus: true
+        });
+        contentState = Modifier.removeRange(contentState, allSelected, 'backward');
+        editorState = EditorState.push(editorState, contentState, 'remove-range');
+        editorState = EditorState.forceSelection(editorState, contentState.getSelectionAfter());
+        setEditorState(editorState);
+    }
 
+    const submitComment = () => {
+        const comment = draftToHtml(
+            rawContentState,
+        )
+        
         addComment(comment, translation);
 
-        commentInput.current.value = '';
-        commentInput.current.rows = minRows;
-    };
+        clearEditorContent();
+    }
+
+    if (!user) {
+        return null;
+    }
 
     return <div className='comment add-comment'>
         <UserAvatar
             username={ username }
             imageUrl={ imageURL }
         />
-        <form className='container'>
-            <Localized
-                id='comments-AddComment--input'
-                attrs={{ placeholder: true }}
-            >
-                <textarea
-                    autoFocus
-                    name='comment'
-                    dir='auto'
-                    placeholder={ `Write a comment…` }
-                    rows={ minRows }
-                    ref={ commentInput }
-                    onChange={ handleOnChange }
-                    onKeyDown={ handleOnKeyDown }
+        <div className='container'>
+            <div className='comment-div' onClick={ focusEditor }>
```

Comment:
Why do we need this wrapper (instead of putting the `onClick` to `Editor` directly)?

Or even: why do we even need to set focus? Shouldn't it be set automatically on click, like in any editor? It also surprises me cursor isn't properly set (to `help`) when you hover the editor before it has focus.

Reply:
The editor gets assigned its own `className` and it does not add a secondary class if one is given. I tried to set the styling to that `className`, but that caused additional issues. So, the only way to get the proper styling is to have the wrapper. 

Removing the `onClick` makes it so that you can't focus on the input area at all, and moving the `onClick` into the `Editor` causes the cursor to disappear if you click anywhere else and it does not refocus when you click back.

Reply:
OK. I'd change the class name to something more semantic, e.g. `.editor-wrapper`.

I noticed that with the following CSS
```css
.DraftEditor-root {
    height: 100%;
    width: 100%;
}
```

I solve a few issues:
- Focus on the editor JustWorks(TM). It's still a good idea to set it onClick, in case user clicks on the padding.
- Cursor is properly set to `cursor: text;`.
- Placeholder no longer requires a [fixed width](https://github.com/mozilla/pontoon/pull/1609#discussion_r406755592).

So I wonder if it's expected for `.DraftEditor-root` to occupy zero space and hence we need to manually set focus, cursor, placeholder width...

Looking at this article, I'd say no:
https://medium.com/@khcarr/styling-the-draft-js-editor-e14152a2472b